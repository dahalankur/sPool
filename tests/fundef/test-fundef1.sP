# functions defined inside functions
def int outer():
        def store bool inner(int i, bool j, mutex m22):
                if (1): 3 else if (2): 4 else 5;;
                return 0;
        return inner;