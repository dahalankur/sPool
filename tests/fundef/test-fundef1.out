PROGRAM {[ 
    FUN(NOSTORE, INTTY, outer, FORMALS(), [ 
        FUN(STORE, BOOLTY, inner, FORMALS((INTTY, i), (BOOLTY, j), (MUTEXTY, m22)), [ 
            IF(INT(1), [ 
                INT(3)], [ 
                IF(INT(2), [ 
                    INT(4)], [ 
                    INT(5)])]) 
            RETURN(INT(0))]) 
        RETURN(VAR(inner))])]}
